src/hooks/useInitialValues.test.ts

```diff
@@ -333,4 +334,239 @@ describe('useInitialValues', () => {
+          concept: '5089AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+          answers: [],
+        },
+        validators: [],
+      },
+      {
+        label: 'BMI:Kg/M2 (Function calcBMI | useFieldValue)',
+        type: 'obs',
+        required: false,
+        id: 'bmi',
+        questionOptions: {
+          rendering: 'number',
+          defaultValue: 0,
+          concept: '1342AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+          isTransient: true,
+          disallowDecimals: false,
+          calculate: {
+            calculateExpression: 'calcBMI(height,weight)',
+          },
+        },
+        validators: [],
+        questionInfo: 'this calculates BMI using calcBMI function and useFieldValue of weight and height',
+      },
+    ];
+    const allFields = JSON.parse(JSON.stringify(formFields));
+    const allFieldsKeys = allFields.map((f) => f.id);
+    let valuesMap = {
+      height: '',
+      wight: '',
+      bmi: '',
+    };
+
+    const helper = new CommonExpressionHelpers(
+      { value: allFields[1], type: 'field' },
+      {},
+      allFields,
+      valuesMap,
+      allFieldsKeys,
+    );
+
+    await act(async () => {
+      hook = renderHook(() =>
+        useInitialValues(
+          [...formFields],
+          encounter,
+          false,
+          {
+            encounter: encounter,
+            patient: testPatient,
+            location,
+            sessionMode: 'enter',
+            encounterDate: encounterDate,
+            setEncounterDate: jest.fn,
+            encounterProvider: '2c95f6f5-788e-4e73-9079-5626911231fa',
+            setEncounterProvider: jest.fn,
+            setEncounterLocation: jest.fn,
+            encounterRole: '',
+            setEncounterRole: jest.fn,
+          },
+          formFieldHandlers,
+        ),
+      );
+    });
+    const {
+      current: { initialValues, isBindingComplete },
+    } = hook.result;
+    expect(isBindingComplete).toBe(true);
+
+    const heightVal = initialValues['height'];
+    const weightVal = initialValues['weight'];
+
+    const calculatedBmi = helper.calcBMI(heightVal, weightVal);
+
+    expect(initialValues['height']).toBe(176);
+    expect(initialValues['weight']).toBe(56);
+    expect(initialValues['bmi']).toBe(calculatedBmi);
+  });
+
+  it('should return asynchronous calculated values for calculated fields in "edit" mode', async () => {
+    let hook = null;
+    const fieldWithCalculateExpression: FormField = {
+      label: 'Latest mother HIV status',
+      type: 'obs',
+      questionOptions: {
+        rendering: 'fixed-value',
+        concept: 'af7c1fe6-d669-414e-b066-e9733f0de7a8',
+        calculate: {
+          calculateExpression:
+            "resolve(api.getLatestObs(patient.id, '159427AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA', '2549af50-75c8-4aeb-87ca-4bb2cef6c69a'))?.valueCodeableConcept?.coding[0]?.code",
+        },
+      },
+      id: 'latest_mother_hiv_status',
+    };
+    allFormFields.push(fieldWithCalculateExpression);
```

Comment:
This mutates the object at the top level of this test. This will lead to behavior that varies depending on which tests are run, or the order in which they are run. Objects at the top level of a test file should be considered immutable. Use the spread syntax to define new form fields within specific tests.